rescript/get_data.py

```diff
@@ -12,13 +12,132 @@
+        '8.0': {'fungi':      {False: '', True: '10.15156/BIO/786349'},
+                'eukaryotes': {False: '', True: ''}},  # All other 8.0 are in zip files
+    }
+    # There's got to be a better way! See https://stackoverflow.com/questions/25833613/safe-method-to-get-value-of-nested-dictionary
+    try:
+        # Check if we have the DOI requested
+        target_doi = unite_dois[version][taxon_group][singletons]
+    except KeyError as ke:
+        print('Unknown DOI for this value: ' + str(ke))
+        raise
+    return _unite_dois_to_urls(target_doi).pop()
+
+_unite_get_url(version='9.0', taxon_group='fungi', singletons=False)
+
+# with tempfile.TemporaryDirectory() as tmp_dir:
+tmp_dir = tempfile.mkdtemp()
+
+def _unite_download_targz(url, download_path):
+    print('Downloading ' + url)
+        
+    response = requests.get(url, stream=True)
+    if response.status_code != 200:
+        raise ValueError("Failed to download the file from " + url)
+    
+    tar_file_path = os.path.join(download_path, 'unitefile.tar.gz')
+    with open(tar_file_path, 'wb') as f:
+        f.write(response.content)
+    
+    # Extract only the 'developer' subdirectory
+    with tarfile.open(tar_file_path, 'r:gz') as tar:
+        # Ensure that 'developer' exists in the tar file
+        members = [member for member in tar.getmembers() if member.name.startswith('developer')]
+        if not members:
+            raise ValueError("No 'developer' subdirectory found in the .tar.gz file.")
+        
+        for member in members:
+            member.name = os.path.basename(member.name) # Strip the 'developer' prefix
+            tar.extract(member, path=download_path)
+    
+    return download_path
+
+# Test it by downloading this file
+# _unite_download_targz('https://files.plutof.ut.ee/public/orig/59/12/591225E8985EFC44B595C79AF5F467421B4D9A95093A0811B13CB4CC13A6DA46.tgz', tmp_dir)
+
+# import as artifacts
+# results[name] = qiime2.Artifact.import_data(dtype, destination)
+
+def get_unite_data(version, taxon_group, singletons=False):
+    url = _unite_get_url(version, taxon_group, singletons)
+    results = {'sequences': [], 'taxonomy': []}
+    
+    # with tempfile.TemporaryDirectory() as tmp_dir:
+    tmp_dir = tempfile.mkdtemp()
+    
+    print('Temporary directory:', tmp_dir)
+    _unite_download_targz(url, download_path=tmp_dir)
+    
+    for root, dirs, files in os.walk(tmp_dir):
+        for file in files:
+            print(results)
+            if file.endswith('.fasta'):
+                fasta_file_name = os.path.join(root, file)
+                print('found fasta: ' + fasta_file_name)
+                with open(fasta_file_name, 'r') as fasta_file:
+                    # Read the content of the file and append it as a Python object
+                    fasta_content = fasta_file.read()
+                    results['sequences'].append(qiime2.Artifact.import_data('FeatureData[RNASequence]', fasta_content))
```

Comment:
is it really RNA? I thought that UNITE used DNA. Some sequences might be mixedcase if I recall... @mikerobeson ?

Reply:
In my other pipeline, I've been importing these as
`--type FeatureData[Sequence] --input-format MixedCaseDNAFASTAFormat`
`--type FeatureData[Taxonomy] --input-format HeaderlessTSVTaxonomyFormat`

I think I'm messing this up in some other way...

Reply:
Yep UNITE is DNA only as far as I've ever seen. But using either `MixedCaseDNAFASTAFormat` or `DNAFASTAFormat` will work. Though I do not recall seening any mixed case data in UNITE, so best be explicit with `DNAFASTAFormat`.

Reply:
In the most recent release, it's all uppercase DNA.

For now, I'll resolve this comment